client/cmdhffelica.c

```diff
@@ -339,12 +306,14 @@ static int CmdHFFelicaRequestService(const char *Cmd) {
     if (all_nodes) {
         for (uint16_t y = 1; y < 32; y++) {
             data[10] = int_to_hex(&y);
-            add_crc_bytes(&datalen, data, sizeof(data));
+            AddCrc(data, datalen);
```

Comment:
you add crc,  but don't increase datalen,   and after sending you remove 2 bytes from datalen...

Reply:
Yeah true, I'm changing the parsing of the raw and request service with the methods from util.c and will test then everything again. Wait with merging please until I've done all the changes.

Reply:
Btw are there any unit tests, I could write?

Reply:
not really... :(

Reply:
Users are the best unit tests anyway ;-D

The util.c helps a lot in parsing. I appreciate the feedback thanks for that! I will refactor the raw command in another pull request - then I will directly add a length parameter to the command.

Reply:
yeah,  we get fast feedback when stuff breaks despite travis, appveyor and coverity  :)